userspace/libsinsp/sinsp.cpp

```diff
@@ -661,22 +699,44 @@ void sinsp::open_savefile(const std::string& filename, int fd)
 	params.start_offset = 0;
 	params.fbuffer_size = 0;
 	oargs.engine_params = &params;
-	open_common(&oargs);
+
+	// AFAICT this is because tinfo==NULL when calling the callback in scap_read_fdlist
+	struct scap_platform* platform = scap_savefile_alloc_platform(nullptr, nullptr);
+	open_common(&oargs, &scap_savefile_engine, platform);
+#else
+	throw sinsp_exception("SAVEFILE engine is not supported in this build");
+#endif
 }
 
 void sinsp::open_plugin(const std::string& plugin_name, const std::string& plugin_open_params, scap_mode_t mode)
 {
+#ifdef HAS_ENGINE_SOURCE_PLUGIN
 	scap_open_args oargs = factory_open_args(SOURCE_PLUGIN_ENGINE, mode);
 	struct scap_source_plugin_engine_params params;
 	set_input_plugin(plugin_name, plugin_open_params);
 	params.input_plugin = &m_input_plugin->as_scap_source();
 	params.input_plugin_params = (char*)m_input_plugin_open_params.c_str();
 	oargs.engine_params = &params;
-	open_common(&oargs);
+
+	struct scap_platform* platform;
+	if(mode == SCAP_MODE_PLUGIN)
```

Comment:
This should mimic old scap.c:
```
if(oargs->mode == SCAP_MODE_LIVE)
		{
			platform = scap_linux_alloc_platform();
		}
		else
		{
			platform = scap_generic_alloc_platform();
		}
```
right?

Reply:
We've already been bitten a couple of times by this :D Just to be sure!

Reply:
We don't expect users to call open_plugin with mode other than LIVE or PLUGIN, so it shouldn't matter either way. Still, now that you pointed it out, I'm inclined to convert this to a switch:

```
case SCAP_MODE_PLUGIN:
    platform = scap_generic_alloc_platform(...);
case SCAP_MODE_LIVE:
    platform = scap_linux_alloc_platform(...);
default:
    ASSERT(false);
    throw ...
```

WDYT? (same for test_input below)

Reply:
@FedeDP fwiw, I did the change to switch{} in a separate commit. Will either fold it into the original one or drop it, based on your feedback

Reply:
i was just thinking out loud when i commented! It's ok as it is :)